sdk/identity/azure-identity/azure/identity/_internal/managed_identity_client.py

```diff
@@ -4,30 +4,29 @@
 # ------------------------------------
 import abc
 import time
-from typing import TYPE_CHECKING, Any, Callable, Dict, Optional, Union
+from typing import Any, Callable, Dict, Optional
 
 from msal import TokenCache
 import six
 
 from azure.core.credentials import AccessToken
 from azure.core.exceptions import ClientAuthenticationError, DecodeError
 from azure.core.pipeline.policies import ContentDecodePolicy
+from azure.core.pipeline import PipelineResponse
+from azure.core.pipeline.transport import HttpRequest
 from .._internal import _scopes_to_resource
 from .._internal.pipeline import build_pipeline
 
-if TYPE_CHECKING:
-    from azure.core.pipeline import PipelineResponse
-    from azure.core.pipeline.policies import HTTPPolicy, SansIOHTTPPolicy
-    from azure.core.pipeline.transport import HttpRequest
 
-    PolicyType = Union[HTTPPolicy, SansIOHTTPPolicy]
-
-ABC = abc.ABC
-
-class ManagedIdentityClientBase(ABC):
+class ManagedIdentityClientBase(abc.ABC):
     # pylint:disable=missing-client-constructor-parameter-credential
-    def __init__(self, request_factory, client_id=None, identity_config=None, **kwargs):
-        # type: (Callable[[str, dict], HttpRequest], Optional[str], Optional[Dict], **Any) -> None
+    def __init__(
+        self,
+        request_factory: Callable[[str, dict], HttpRequest],
+        client_id: Optional[str] = None,
+        identity_config: Optional[Dict] = None,
```

Comment:
Can the Dict type be narrowed? If not that's fine, mostly just wondering since I've seen a few Dict[Any] type hints across this PR

Reply:
The detailed structure of identity_config is still TBD.  :)